core-management/core-management-subsystem/src/test/java/org/wildfly/extension/core/management/CoreManagementSubsystemTestCase.java

```diff
@@ -32,4 +57,17 @@ protected AdditionalInitialization createAdditionalInitialization() {
         return new AdditionalInitialization.AdminOnlyHostControllerAdditionalInitialization(getSubsystemSchema());
     }
 
+    @Test
+    public void testExpressionAttributesResolved() {
+        ModelNode processStateListener = serverModel.get(PROCESS_STATE_LISTENER).get("x");
+        ModelNode processStateListenerProperties = processStateListener.get(PROPERTIES);
+
+        assertEquals("5000", getValue(processStateListener, TIMEOUT));
+        assertEquals("2", getValue(processStateListenerProperties, "bar"));
+    }
+
+    private Object getValue(ModelNode node, String attributeName) {
+        ModelNode result = node.get(attributeName).resolve();
+        return result.asString();
+    }
```

Comment:
Reviewing this I've just learnt that Elytron tests (e.g ResolveExpressionAttributesTestCase) use the same approach you did here. However, this is a weak test. What we are testing here is just how the ModelNode resolves expressions.

What I was expecting instead is to read the resource with the expressions resolved so we can getin the operation result the expressions resolved by the server code, hence we will be testing the expression resolution of the read operation with this subsystem instead of just testing that the ModelNode object is able to resolve an expression.

@lvydra Do you follow me on this and do you agree? For your info, I've found this example of what we should be testing here https://github.com/wildfly/wildfly/blob/main/naming/src/test/java/org/jboss/as/naming/subsystem/NamingSubsystemTestCase.java#L113-L117

 I'm going to open a Jira for ResolveExpressionAttributesTestCase since in my opinion it should be hardened testing the server code expression resolution and not the ModelNode object.

Reply:
@yersan Now that I see it, I have to agree. I'll look into it.